src/pages/spawn/spawnHost/SpawnHostTable.tsx

```diff
@@ -1,3 +1,149 @@
+  copyToClipboard,
+  sortFunctionDate,
+  sortFunctionString,
+} from "utils/string";
+import { SpawnHostActionButton } from "./SpawnHostActionButton";
 
-export const SpawnHostTable = () => <>Spawn Host table</>;
+interface SpawnHostTableProps {
+  hosts: Host[];
+}
+export const SpawnHostTable: React.FC<SpawnHostTableProps> = ({ hosts }) => {
+  const columns = [
+    {
+      title: "Host",
+      dataIndex: "id",
+      key: "host",
+      sorter: (a, b) => sortFunctionString(a, b, "id"),
+      render: (_, host: Host) => <HostIdField host={host} />,
+    },
+    {
+      title: "Distro",
+      dataIndex: "distro",
+      key: "distro",
+      sorter: (a, b) => sortFunctionString(a, b, "distro.id"),
+      render: (distro) => distro.id,
+    },
+    {
+      title: "Status",
+      dataIndex: "status",
+      key: "status",
+      sorter: (a, b) => sortFunctionString(a, b, "status"),
+      render: (status) => <HostStatusBadge status={status} />,
+    },
+    {
+      title: "Expires In",
+      dataIndex: "expiration",
+      key: "expiration",
+      sorter: (a, b) => sortFunctionDate(a, b, "expiration"),
+      render: (expiration) =>
+        formatDistanceToNow(new Date(expiration)) || "Does not expire",
+    },
+    {
+      title: "Uptime",
+      dataIndex: "uptime",
+      key: "uptime",
+      sorter: (a, b) => sortFunctionDate(a, b, "uptime"),
+      render: (uptime) => formatDistanceToNow(new Date(uptime)),
+    },
+    {
+      title: "Action",
+      key: "action",
+      render: (_, host) => <SpawnHostActions host={host} />,
+    },
+  ];
+  return (
+    <Container>
+      <Table
+        columns={columns}
+        dataSource={hosts}
+        rowKey={(record) => record.id}
+        pagination={false}
+      />
+    </Container>
+  );
+};
+
+const SpawnHostActions: React.FC<{ host: Host }> = ({ host }) => (
+  <ActionButtonContainer>
+    <SpawnHostActionButton host={host} />
+    <PaddedButton glyph={<Icon glyph="Trash" />} size={Size.XSmall} />
+    <CopySSHCommandButton host={host} />
+    <PaddedButton size={Size.XSmall}>Edit</PaddedButton>
+  </ActionButtonContainer>
+);
+
+const CopySSHCommandButton: React.FC<{ host: Host }> = ({ host }) => {
+  const sshCommand = `${host.user}@${host.hostUrl}`;
+
+  return (
+    <Tooltip
+      align="top"
+      justify="middle"
+      trigger={
+        <PaddedButton
+          onClick={() => copyToClipboard(sshCommand)}
+          size={Size.XSmall}
+        >
+          Copy SSH command
+        </PaddedButton>
+      }
+      triggerEvent="click"
+      variant="light"
+    >
+      Copied!
+    </Tooltip>
+  );
+};
+
+const HostIdField: React.FC<{ host: Host }> = ({ host }) => {
+  const isVirtualWorkStation = host?.distro?.isVirtualWorkStation;
+  return isVirtualWorkStation ? (
+    <FlexContainer>
+      <HostIdSpan>{host.id}</HostIdSpan>
+      <WorkstationBadge>WORKSTATION</WorkstationBadge>
+    </FlexContainer>
+  ) : (
+    <HostIdSpan>{host.id}</HostIdSpan>
+  );
+};
```

Comment:
seems overkill to have this be its own component

Reply:
I dont think so, I think its cleaner this way  then trying to inline the logic inside of the render method of the columns field of the spawn host table

Reply:
if that's the case, then if it deserves to be its own component then it should be its own file too